src/Microsoft.ApiDesignGuidelines.Analyzers/Core/ParameterNamesShouldMatchBaseDeclaration.cs

```diff
@@ -29,10 +34,113 @@ public abstract class ParameterNamesShouldMatchBaseDeclarationAnalyzer : Diagnos
 
+        private static readonly Action<SymbolAnalysisContext> AnalyzeMethodSymbolAction = AnalyzeMethodSymbol;
+
+        /// <inheritdoc/>
         public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(Rule);
 
+        /// <inheritdoc/>
         public override void Initialize(AnalysisContext analysisContext)
         {
+            analysisContext.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.None);
+            analysisContext.EnableConcurrentExecution();
+
+            analysisContext.RegisterSymbolAction(AnalyzeMethodSymbolAction, SymbolKind.Method);
+        }
+
+        private static void AnalyzeMethodSymbol(SymbolAnalysisContext analysisContext)
+        {
+            var methodSymbol = analysisContext.Symbol as IMethodSymbol;
+
+            if (!(methodSymbol.CanBeReferencedByName || methodSymbol.IsImplementationOfAnyExplicitInterfaceMember())
+                || !methodSymbol.Locations.Any(x => x.IsInSource)
+                || string.IsNullOrWhiteSpace(methodSymbol.Name))
+            {
+                return;
+            }
+
+            if (!methodSymbol.IsOverride && !methodSymbol.IsImplementationOfAnyImplicitInterfaceMember())
+            {
+                return;
+            }
+
+            ImmutableArray<IMethodSymbol> originalDefinitions = GetOriginalDefinitions(methodSymbol);
+            if (originalDefinitions.Length == 0)
+            {
+                // We did not find any original definitions so we don't have to do anything.
+                // This can happen when the method has an override modifier,
+                // but does not have any valid method it is overriding.
+                return;
+            }
+
+            IMethodSymbol bestMatch = null;
+            int bestMatchScore = -1;
+
+            foreach (var originalDefinition in originalDefinitions)
+            {
+                // always prefer the method override, if it is available
+                // (the overridden method will always be the first item in the list.)
+                if (originalDefinition.ContainingType.TypeKind != TypeKind.Interface)
+                {
+                    bestMatch = originalDefinition;
+                    break;
+                }
+
+                int currentMatchScore = 0;
+                for (int i = 0; i < methodSymbol.Parameters.Length; i++)
+                {
+                    IParameterSymbol currentParameter = methodSymbol.Parameters[i];
+                    IParameterSymbol originalParameter = originalDefinition.Parameters[i];
+
+                    if (currentParameter.Name == originalParameter.Name)
+                    {
+                        currentMatchScore++;
+                    }
+                }
+
+                if (currentMatchScore > bestMatchScore)
+                {
+                    bestMatch = originalDefinition;
```

Comment:
I like your approach of suggesting the best match by count of matching parameter names. FXCop actually searches through the list of original definitions and reports an issue for each definition against which the match fails - that seems like lot of clutter.